telluric/georaster.py

```diff
@@ -206,27 +207,24 @@ def _merge(one, other, merge_strategy=MergeStrategy.UNION):
         other_remaining_bands = [band for band in other.band_names if band not in set(common_bands)]
 
         if one_remaining_bands:
-            all_data.insert(0, one.limit_to_bands(one_remaining_bands).image.data)
+            all_data.insert(0, one.subimage(one_remaining_bands).data)
             # This is not necessary, as new_mask already includes
             # at least all the values of one.image.mask because it comes
             # either from one or from the intersection of one and other
             # new_mask |= one.image.mask[0]
             new_bands = one_remaining_bands + new_bands
 
         if other_remaining_bands:
-            all_data.append(other.limit_to_bands(other_remaining_bands).image.data)
+            all_data.append(other.subimage(other_remaining_bands).data)
             # Apply "or" to the mask in the same way rasterio does, see
             # https://mapbox.github.io/rasterio/topics/masks.html#dataset-masks
             new_mask |= other.image.mask[0]
             new_bands = new_bands + other_remaining_bands
-
         new_image = np.ma.MaskedArray(
             np.concatenate(all_data),
-            mask=new_mask[None].repeat(len(new_bands), axis=0) == 1
+            mask=[new_mask]*len(new_bands)
         )
-
-        return res_common.copy_with(image=new_image, band_names=new_bands)
-
+        return Raster(image=new_image, band_names=new_bands)
```

Comment:
I think here you should do `return Raster(image=deepcopy(new_image), band_names=deepcopy(new_bands))` o breake reference to `one` and `other` objects

Reply:
why? they are created in the lines before